### DISCUSSION TITLE
[Idea] Add suport for retrieving complete Windows build version

### DISCUSSION THREAD

**Discussion by Carlos-Borau** [glpi-project/glpi-agent#410] [Ideas]
### Your idea

So basicaly I've been using the GLPI agent to pull OS versions from multiple Windows computers and trying to use the inventory output to search for CVEs related to the retrieved OS versions. My problem is that the agent only seems to pull part of the build number as the kernel version, so for example, let's say I have a computer running Windows 10 Home 22h2 with the OS build being 19045.2965. When running glpi-inventory --json on this machine, the following OS attributes will be retrieved:

-"full_name": "Microsoft Windows 10 Home"
-"version": "22H2"
-"kernel_name": "MSWin32"
-"kernel_version": "10.0.19045"

If i wanted to compose a CPE WFN to search vulnerabilities related to this particular OS installation i would need it to look like this: "cpe : 2.3 : o : microsoft : windows_10_22h2 : 10.0.19045.2965". I found that the remaining number (2965) is stored on the "\HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows NT\CurrentVersion\UBR" key, so it should definitely be possible to get the full build version when running the inventory agent.

I couldn't find any discussion related to this topic so I thought it would be interesting to bring it up.
Is there a way to alrady do this?
If yes, how?
If not, would this be something interesting to implement?

**Comment by g-bougard** [comment]
Hi @Carlos-Borau 

but a question related to that subject has still been asked few days ago in #408.

Using nightly build, you should be able to find the UBR number into the SERVICE_PACK field.
